modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsNorth.scala

```diff
@@ -0,0 +1,172 @@
+import squants.Angle
+import squants.space.Arcseconds
+
+class TcsNorth[F[_]: Sync] private (tcsController: TcsNorthController[F],
+                                    subsystems: NonEmptySet[Subsystem],
+                                    gaos: Option[Altair[F]],
+                                    guideDb: GuideConfigDb[F]
+                                   )(config: TcsNorth.TcsSeqConfig[F]) extends System[F] {
+  import TcsNorth._
+  import Tcs.{GuideWithOps, calcGuiderInUse}
+
+  override val resource: Resource = Resource.TCS
+
+  // Helper function to output the part of the TCS configuration that is actually applied.
+  private def subsystemConfig(tcs: TcsNorthConfig, subsystem: Subsystem): List[String] = subsystem match {
+    case Subsystem.M1     => List(tcs.gc.m1Guide.show)
+    case Subsystem.M2     => List(tcs.gc.m2Guide.show)
+    case Subsystem.OIWFS  => List((tcs.gds.oiwfs:GuiderConfig).show)
+    case Subsystem.PWFS1  => List((tcs.gds.pwfs1:GuiderConfig).show)
+    case Subsystem.PWFS2  => List(tcs.gds.pwfs2OrAowfs.swap.getOrElse(
+      GuiderConfig(ProbeTrackingConfig.Parked, GuiderSensorOff)).show)
+    case Subsystem.Mount  => List(tcs.tc.show)
+    case Subsystem.AGUnit => List(tcs.agc.sfPos.show, tcs.agc.hrwfs.show)
+    case Subsystem.Gaos   => List(tcs.gds.pwfs2OrAowfs.toOption).collect{ case Some(x) => (x:GuiderConfig).show }
+  }
+
+  override def configure(config: Config): SeqActionF[F, ConfigResult[F]] = SeqActionF.embedF(
+    buildTcsConfig.flatMap{ cfg =>
+      Log.debug(s"Applying TCS configuration: ${subsystems.toList.flatMap(subsystemConfig(cfg, _))}").pure[F] *>
+        tcsController.applyConfig(subsystems, gaos, cfg).as(ConfigResult(this))
+    }
+  )
+
+  override def notifyObserveStart: SeqActionF[F, Unit] = SeqActionF.embedF(tcsController.notifyObserveStart)
+
+  override def notifyObserveEnd: SeqActionF[F, Unit] = SeqActionF.embedF(tcsController.notifyObserveEnd)
+
+  val defaultGuiderConf = GuiderConfig(ProbeTrackingConfig.Parked, GuiderSensorOff)
+  def calcGuiderConfig(inUse: Boolean, guideWith: Option[StandardGuideOptions.Value]): GuiderConfig =
+    guideWith.flatMap(v => inUse.option(GuiderConfig(v.toProbeTracking, v.toGuideSensorOption)))
+      .getOrElse(defaultGuiderConf)
+
+  private val usesAltair = gaos.isDefined
+  /*
+   * Build TCS configuration for the step, merging the guide configuration from the sequence with the guide
+   * configuration set from TCC. The TCC configuration has precedence: if a guider is not used in the TCC configuration,
+   * it will not be used for the step, regardless of the sequence values.
+   */
+  def buildTcsConfig: F[TcsNorthConfig] =
+    guideDb.value.map{ c => {
+      val useAo: Boolean = c.gaosGuide match {
+        case Some(Left(AltairOff)) => false
+        case Some(Left(_))         => true
+        case _                     => false
+      }
+      val aoUsesP1 = useAo && (gaos, c.gaosGuide.flatMap(_.swap.toOption)).mapN(_.usesP1(_))
```

Comment:
`gaosGuide` is optional, and if present, it could be an Altair or GeMS configuration. Its type is `Option[Either[AltairConfig, GemsConfig]]`
For this case, we only care if it is Altair. The line `gaosGuide.flatMap(_.swap.toOption)` transforms the `Option[Either[AltairConfig, GemsConfig]]` into a `Option[AltairConfig]`, where the case `Some[GemsConfig]` is collapsed into a `None`. Your suggestion would make it a `Option[Either[GemsConfig, AltairConfig]]`